dali/test/python/test_functional_api.py

```diff
@@ -0,0 +1,79 @@
+import nvidia.dali.fn as fn
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.types as types
+import numpy as np
+
+def _test_fn_rotate(device):
+    pipe = Pipeline(batch_size = 1, num_threads = 1, device_id = 0)
+
+    image = np.array([
+        [1, 2,  3,  4],
+        [5, 6,  7,  8],
+        [9, 10, 11, 12]], dtype=np.uint8)[:,:,np.newaxis]
+    batch = [image]
+
+    input = fn.external_source([batch], layout = "HWC")
```

Comment:
```suggestion
    input = fn.external_source(batch, layout = "HWC")
```
It's already a list of np.arrays.

Reply:
We need to have a _collection of batches_, not a batch.

Reply:
Oh. That's a bit tricky. How would the error look if we pass a batch. Looking at what I assumed, I'm guessing it will pop up few times in the issues.

Reply:
In this case? The error would say that "HWC" cannot be used to describe 2D data. If the outermost dimension of the batch item coincides with the batch size _and_ there is no layout, there will be no error at all.